core/parentchain/block-import-dispatcher/src/immediate_dispatcher.rs

```diff
@@ -17,16 +17,28 @@
 
 use crate::{error::Result, DispatchBlockImport};
 use itc_parentchain_block_importer::ImportParentchainBlocks;
-use std::{sync::Arc, vec::Vec};
+use log::*;
+use std::{boxed::Box, vec::Vec};
 
 /// Block import dispatcher that immediately imports the blocks, without any processing or queueing.
 pub struct ImmediateDispatcher<BlockImporter> {
-	block_importer: Arc<BlockImporter>,
+	block_importer: BlockImporter,
+	import_event_observers: Vec<Box<dyn Fn() + Send + Sync + 'static>>,
```

Comment:
Not according to our compiler, unfortunately:
```
error[E0277]: the size for values of type `(dyn Fn() + Send + Sync + 'static)` cannot be known at compilation time
   --> /home/femueller/repos/remote/integritee-worker/core/parentchain/block-import-dispatcher/src/immediate_dispatcher.rs:26:26
    |
26  |     import_event_observers: Vec<dyn Fn() + Send + Sync + 'static>,
    |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ doesn't have a size known at compile-time
    |
    = help: the trait `Sized` is not implemented for `(dyn Fn() + Send + Sync + 'static)`
```

Reply:
Ah, no sorry. This was because I told you that I believe you can only use one concrete type in the `Vec` even if you use the `dyn` pattern, but it seems that you were right that we can do a vec with:

```
let dog: Dog = Dog::new();
let cat: Cat = Cat::new();

let vec: Vec<Box<dyn Animal>> = vec![Box::new(dog), Box::new(cat)];
```

if both, `Dog` and `Cat` implement `Animal`.

Reply:
Ah okay, then we're on the same page 😄